lib/std/special/docs/main.js

```diff
@@ -414,6 +426,8 @@
                 }
             }
 
+            domListPkgs.innerHTML = "";
+            domListPkgs.appendChild(domListPkgsFragment);
```

Comment:
Use [`replaceChild`](https://developer.mozilla.org/en-US/docs/Web/API/Node/replaceChild) instead of clearing and appending.

Reply:
`replaceChild` does not clear the subtree though, and domListPkgs has many children. i may be missing something obvious but i do not see how replaceChild brings a simpler solution to this ?

Reply:
Instead of using document fragments and `<template>` elements, the `domListPkgsFragment` (with a different name, presumably) could just be another `<ul>` element, created with `domListPkgs.cloneNode(false)`. The update would then do
```suggestion
            domListPkgs.parentNode.replaceChild(domListPkgsFragment, domListPkgs);
            domListPkgs = domListPkgsFragment;
```
Since the new list is created with a clone, it has all the right attributes, but since the clone is shallow, none of the children are copied. This has the advantage of significantly larger and older browser support (though I don't know what Zig wants to support), as well as only doing one DOM update instead of two (in addition to performance, this could theoretically reduce flashing).

Reply:
Oh yes I have thought about that, but given the actual design it would have obliged me to make sure to reassign the globals referring to the parent node (in this case `var domListSearchResults = document.getElementById("listSearchResults");`) and potential event listeners and so on... I also wanted my changes to be minimal.

I'm not sure what exact issue we are trying to solve here, but I have measured no difference between `clear innerHTML + appendChild` and `replaceChild`, and seen no flashing.

I still do not really see what replaceChild brings to the table here, but I'm willing to make the necessary changes so that we can actually compare. Will try to do it soon.

Reply:
Sorry, to be clear, I saw that as a possible alternative that might be simpler and faster, but I don't have strong opinions and haven't benchmarked or tested anything. The main motivation to change would be IE support, and I don't know if that is desired.

I didn't realize that `cloneNode` didn't copy event listeners; that's unfortunate, and makes it a lot less appealing, even though there currently aren't any event listeners on list-like elements.